WPMedia/fusion-news-theme-blocks
# TBRANDS-28 => Story Carousel

## Description

The birth of the Story Carousel Block

## Jira Ticket

- [[TBRANDS-25](https://arcpublishing.atlassian.net/browse/TBRANDS-25)](https://arcpublishing.atlassian.net/browse/TBRANDS-28)

## Something to note:  

While I don't consider this urgent, when you view this, you will see that on initial render, the underlying carousel component does a wonky re render as its figures out the layout.  Thinking we should look into some kind of callback from the carousel component that lets us know it has rendered correctly.  We could then hide the carousel until we get that callback letting us know we can show it.

## Acceptance Criteria

As a PageBuilder Editor user:
I can add a Story Carousel Block to my page and then configure that block to run off a WebSked collection of Composer Stories (ANS feed) by selecting the WebSked collection from within Page Builder Editor. 

I can input the Carousel heading via a manual custom field in Page Builder Editor (optional field).
If there are less than 4 “Story Cards” alert the user that “This feature requires 4 stories to display.”

Display Logic for Headline and Description:

Headline: We want to limit the headline to 2 rows. If the Headline content expands more than 2 rows we want to truncate the Headline and add an ellipses (…) to the end of the second line. 

Description: We want to limit the description to 3 rows. If the Headline content expands more than 3 rows we want to truncate the Headline and add an ellipses (…) to the end of the third line. 

## Test Steps

1. Checkout this branch `git checkout {branch name}`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/story-carousel-block,@wpmedia/single-column-layout-block,@wpmedia/category-carousel-block`
3. Open Storybook: `npm run storybook`
4. To run locally, configure your .env file to use the Arc Commerce Sandbox environment.
5. Run local fusion after pulling other PR: `npx fusion start -f -l @wpmedia/story-carousel-block,@wpmedia/single-column-layout-block`
6. Give the block a headline, a value of 2 for headline truncation and a value of 3 for description truncation. ( I think these should be props and not styles as each story carousel may want a different number of lines).
7. Fill block using collection content source....`homepage` is a good enough tag to pull in stories.

## Dependencies or Side Effects

https://github.com/WPMedia/arc-themes-feature-pack/pull/220

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

- [ x] Confirmed all the test steps a reviewer will follow above are working.
- [x ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x ] Confirmed this PR has unit test files
  - [ x] Ran `npm run test`, made sure all tests are passing
  - [x ] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._
